### DISCUSSION TITLE
Can we develop ourselves JavaScriptServlet?

### DISCUSSION THREAD

**Discussion by zhengjiudan** [OWASP/www-project-csrfguard#113] [General]
Currently we are suing version 3.1 where we use JavaScriptServlet::doPost to fetch the token. But doPost changes a lot on version 4.1. As we don't want big change on front end, we want to introduce another Servlet whose doPost returns the master token. I just want to confirm if this solution will introduce any vulnerability. From my perspective, as doGet of new version Servlet returns MasterToken as well, I don't see any vulnerability of letting doPost to return MasterToken.

I understand CSRFGuard 4.1 is not to store token in HTTP session any more which is to prevent XSS attack. Is this the case?

**Comment by forgedhallpass** [comment]
Hello @zhengjiudan,

The upgrade from `3.x` to `4.x` should be seamless. You shouldn't need to modify any code.

Please refer to the other related discussions:
* https://github.com/OWASP/www-project-csrfguard/discussions/91#discussioncomment-2607877
* https://github.com/OWASP/www-project-csrfguard/discussions/52#discussioncomment-1923833
* https://github.com/OWASP/www-project-csrfguard/discussions/51#discussioncomment-1862899
* https://github.com/OWASP/www-project-csrfguard/discussions/98#discussioncomment-2665521
* https://github.com/OWASP/www-project-csrfguard/discussions/102#discussioncomment-2789536
